core/server/web/api/canary/admin/routes.js

```diff
@@ -220,6 +220,9 @@ module.exports = function apiRoutes() {
     router.del('/invites/:id', mw.authAdminApi, http(apiCanary.invites.destroy));
 
     // ## Redirects (JSON based)
+    // TODO: yaml support has been added to https://github.com/TryGhost/Ghost/issues/11085
```

Comment:
This is a `canary` route and it hasn't been marked as "stable" anywhere iirc. Would need to give a thought about renaming this endpoint in current version :thinking:

Reply:
I love your idea of naming them `redirects/upload`, `redirects/download`. They're more intuitive. 

But I'm a bit against removing current `redirects/json` APIs. Because it's a breaking change and it's against the semver rule.

My suggestion is to just add `redirects/upload`, `redirects/download` and remove `redirect/json` in v4.

Reply:
Think your suggestion is the best way forward :+1: We'd need to switch over Ghost-Admin to use the new endpoint once these changes are merged. Can you add a note in the main issue or create a Ghost-Admin PR linking this this issue please? This way both changes can go into master together.